src/main/python/pybuilder/plugins/python/coverage_plugin.py

```diff
@@ -94,14 +117,18 @@ def do_coverage(project, logger, reactor, execution_prefix, execution_name, targ
     sum_lines = 0
     sum_lines_not_covered = 0
 
-    module_names = _discover_modules_to_cover(project)
     modules = []
     for module_name in module_names:
         try:
             module = sys.modules[module_name]
         except KeyError:
-            logger.warn("Module not imported: {0}. No coverage information available.".format(module_name))
-            continue
+            logger.warn("Module '%s' was not imported by the covered tests", module_name)
+            try:
+                module = __import__(module_name)
+            except SyntaxError as e:
+                logger.warn("Coverage for module '%s' cannot be established - the module doesn't compile: %s",
```

Comment:
:+1: but maybe "compile" is the wrong wording to use - it's more like unparseable than not compilable